src/shared/components/case-editor/services/wizard-page-field-to-case-field.mapper.ts

```diff
@@ -0,0 +1,84 @@
+import { WizardPageField } from '../domain';
+import { CaseField } from '../../../domain';
+import { ComplexFieldMask } from '../domain/wizard-page-field-complex-mask.model';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class WizardPageFieldToCaseFieldMapper {
+
+  mapAll(wizardPageField: WizardPageField[], caseFields: CaseField[]): CaseField[] {
+    return wizardPageField.map((wizardField: WizardPageField) => {
+      return this.map(wizardField, caseFields);
+    });
+  }
+
+  private map(wizardPageField: WizardPageField, caseFields: CaseField[]): CaseField {
+
+    let case_field: CaseField = caseFields.find(e => e.id === wizardPageField.case_field_id);
+    case_field.id = wizardPageField.case_field_id;
+    case_field.wizardProps = wizardPageField;
+    case_field.display_context = wizardPageField.display_context;
+    case_field.order = wizardPageField.order;
+
+    if (wizardPageField.display_context === 'COMPLEX' && wizardPageField.complex_field_mask_list) {
+      wizardPageField.complex_field_mask_list.forEach((complexFieldMask: ComplexFieldMask) => {
+        this.processComplexFieldMask(complexFieldMask, case_field, caseFields);
+      });
+    }
+
+    return case_field;
+  }
+
+  private processComplexFieldMask(complexFieldMask: ComplexFieldMask, case_field: CaseField, caseFields: CaseField[]) {
```

Comment:
can we rename `complexFieldMask` to `mask`, or `override` (according to the comments above)